__init__.py

```diff
@@ -552,8 +570,29 @@ def _get_alarm_matches(
             time_alarm = to_utc(when).timestamp()
             if is_midnight:
                 time_alarm = time_alarm + 86400.0
+
             time_matches = [a for a in alarms if abs(a["timestamp"] - time_alarm) <= 60]
 
+        # add other categories of alarm matches here
+        utt = self.workaround_lingua_franca(utt)
+        when, utt_no_datetime = extract_datetime(utt) or (None, utt)
+
+        when_utc = None
+        if when is not None:
+            when_utc = to_utc(when)
```

Comment:
could be simplified into `when_utc = None if when is None else to_utc(when)`

Reply:
Done